x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/util/QueryPage.java

```diff
@@ -38,7 +37,7 @@
     public QueryPage(List<T> results, long count, ParseField resultsField) {
```

Comment:
devils advocate: as we are moving this into a more prominent place, we should discuss the naming of this, I think `QueryPage` doesn't fit. It's rather a `ResultPage`.

Reply:
Hmm, definitely a possibility. I think the name `QueryPage` is an artifact of indicating that "This is the results of a Query", but `ResultPage` does give a broader indication as there is nothing restricting the class.

Reply:
@dimitris-athanasiou ^^ what say you?

Reply:
I would even say it is just a `Page` :-) However, changing this might cause BWC problems, at least on the HLRC and transport level.

Reply:
Yes, this change is already breaking for transport client users (so I added the `>breaking-java` label).  To move a class from one package to another is not too hard to cope with for a developer with an IDE, as the IDE will find the class of the same name in its new place.  But to change both the package and the name of a class in a refactoring is quite unfriendly.  Plus if the class is renamed server-side then the HLRC equivalent should be renamed to match, and that means we cause work for HLRC users as well as transport client users (as well as extra work for ourselves changing the HLRC docs).  So I am also of the opinion that we should stick with the current name even if it isn't perfect.

Reply:
I do not see (a) `QueryPage` being used in HLRC.

Reply:
> I do not see (a) `QueryPage` being used in HLRC.

Yes, that's true.  The HLRC responses just contain Java `List`s.  So we didn't maintain the interface for response objects between the transport client and the HLRC.  I thought we had.  Sorry.

Reply:
True, we've already stepped away from it there with the `AbstractResultResponse`. I'm glad we did that :-)